src/dstack/_internal/server/services/fleets.py

```diff
@@ -395,6 +398,9 @@ async def create_fleet_instance_model(
 ) -> InstanceModel:
     profile = spec.merged_profile
     requirements = _get_fleet_requirements(spec)
+    blocks: Optional[Union[Literal["auto"], int]] = spec.configuration.blocks
+    if isinstance(blocks, int) and blocks < 2:
+        blocks = None
```

Comment:
Setting shared_info to None when `blocks: 1` is very confusing since you need to know that total_blocks cannot be 1 (e.g. when shared offers are generated). I'd reverse it: if blocks is None, then assume `blocks: 1`.

Reply:
Again, with my solution, it would be more appropriate to normalize shared_info to keep total/used blocks in separate columns so that it's easier to migrate.

Reply:
Refactor as part of https://github.com/dstackai/dstack/pull/2253/commits/8b555f7a6ad65afbed244e3d1e76359bdbbcbbf9